app/uk/gov/hmrc/preferenceschangednotifier/controllers/PreferencesChangedController.scala

```diff
@@ -33,21 +33,23 @@ class PreferencesChangedController @Inject() (
 )(implicit val ec: ExecutionContext)
     extends BackendController(cc) {
 
-  implicit val pcr: Reads[PreferencesChangedRequest] =
-    PreferencesChangedRequest.reads
+  import PreferencesChangedRequest.given_Reads_PreferencesChangedRequest
 
-  def preferencesChanged(): Action[JsValue] = Action.async(parse.json) { implicit request =>
-    withJsonBody[PreferencesChangedRequest] { body =>
-      svc
-        .preferenceChanged(body)
-        .fold(
-          {
-            case RequestError(r)     => BadRequest(r)
-            case PersistenceError(p) => InternalServerError(p)
-            case ServerError(s)      => InternalServerError(s)
-          },
-          _ => Ok
-        )
+  def preferencesChanged(): Action[JsValue] = Action.async(parse.tolerantJson) { implicit request =>
+    request.body.validate[PreferencesChangedRequest] match {
+      case JsSuccess(body, _) =>
+        svc
+          .preferenceChanged(body)
+          .fold(
+            {
+              case RequestError(r)     => BadRequest(r)
+              case PersistenceError(p) => InternalServerError(p)
```

Comment:
suggestion -  I think you can combine the last 2 error types as they are giving same response.

Reply:
Multiple matches cannot be bound, so in my mind its not quite so clean, but happy to take your suggestion